packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -135,9 +135,16 @@ export function getDeserializePrivateFunction(
   if (needUnexpectedHelper) {
     statements.push(
       `if(${needSubClient ? "UnexpectedHelper." : ""}isUnexpected(result)){`,
-      "throw result.body",
+      `throw createRestError(result);`,
       "}"
     );
+    const specifier = getImportSpecifier("restClient", runtimeImports);
```

Comment:
nit: feels like we should have an "addImportSpecifier" helper that is smart enough to check if it exists and to populate the importSet